Podstawy programowania/Ćwiczenia powtórzeniowe/39-R_Cwiczenie_powtorzeniowe3.py

```diff
@@ -0,0 +1,54 @@
+# Zadanie 3
+# Wczytać długości 3 boków trójkąta. Sprawdzić czy z podanych trzech długości odcinków
+# można zbudować trójkąt, jeżeli tak obliczyć jego pole.
+# P=sqrt(p*(p-a)*(p-b)*(p-c)),
+# gdzie p jest połową obwodu.
+import math
+
+def get_side(name):
+    number = input("Podaj długość boku " + name + ": ")
+    while True:
+      try:
+          number = float(number)
+          if number <= 0:
+              number = input("Długość boku trójkąta nie może być mniejsza lub równa 0. \nPodaj jeszcze raz długość boku " + name + " ")
+          else:
```

Comment:
Tutaj też chyba możnaby to zrobić nieco inaczej:

```
    while True:
      try:
          number = float(number)
          if number <= 0:
              number = input("Długość boku trójkąta nie może być mniejsza lub równa 0. \nPodaj jeszcze raz długość boku " + name + " ")
          else:
            return number
      except:
          number = input("Błąd, podaj liczbę: ")
```

Reply:
W międzyczasie poprawiłem chyba, bo jest tak jak napisałaś...